src/targets/gpu/compile_hip.cpp

```diff
@@ -251,8 +251,12 @@ compile_hip_src(const std::vector<src_file>& srcs, std::string params, const std
             std::cout << std::string(src.content) << std::endl;
         }
     }
-    auto p      = dynamic_loader::path(&compile_hip_src_with_hiprtc);
+    auto p = dynamic_loader::path(&compile_hip_src_with_hiprtc);
+#ifdef _WIN32
+    auto driver = p.parent_path() / "migraphx-hiprtc-driver.exe";
```

Comment:
We do not use the same "installation" path on Windows as on Linux. Don't be suggested on what `make install` does. Both DLL and the driver need to be in the same directory, and the name for the parent directory is arbitrary - `bin` is not the requirement.

Reply:
This is definitely an area where Windows and Linux diverge. Linux typically installs things in set paths on a system whereas applications on Windows typically package their dependencies on the same folder and expect to find their dependencies there. It's not to say that it is impossible to do otherwise, but it is a very non-standard way of doing things.

I've added some screenshots of 3 random apps on my machine installed in Program Files to show this pattern. As a counter example to myself, I included git on Windows which actually follows something closer to the pattern you are suggesting, but I'd prefer if we follow the standard way of doing things and not force a bin search path. 

Following the pattern (same directory):

![image](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/assets/30125873/7abbe2bd-903c-4faa-a3d4-5ec2714cbd2a)

![image](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/assets/30125873/b20c441f-129e-4346-a397-5b6c2ce19608)

![image](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/assets/30125873/edea784e-8b0a-47ff-9924-57cd50ea6949)

Not following the pattern (bin directory):

![image](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/assets/30125873/c0821b19-7ae8-4630-a2c8-94b89ae894a9)

Reply:
> We do not use the same "installation" path on Windows as on Linux.

With the way the current rocm-cmake works, yes we do.

> Don't be suggested on what make install does.

It must use our install script, we arent going to support frankenstein installations upstream(users can make changes to the source code if they want such alternative installations). 

> Both DLL and the driver need to be in the same directory, and the name for the parent directory is arbitrary - bin is not the requirement.

The `bin` directory is what we currently support, but we are going to be moving this into a private directory(such as `libexec`) in the future because this is not a user facing executable so it shouldn't be on the system path.

Reply:
Abstracting from the above, on Windows, `migraphx-hiprtc-driver` != `migraphx-hiprtc-driver.exe`.
Just `migraphx-hiprtc-driver` is ambiguous on Windows. It could be executable (EXE), batch file (BAT), command file (COM), command script (CMD), ShellScript (PS1), and more.
 As Tom mentioned, here is the place where Windows and Linux diverge. Going Linux way here will complicate deployment on Windows a lot.